5.x/crud-operation-trash.md

```diff
@@ -0,0 +1,154 @@
+# Trash Operation <span class="badge badge-info">PRO</span>
+
+--
+
+<a name="about"></a>
+## About
+
+This CRUD operation allows your admins to soft delete, restore and permanently delete entries from the table.
+
+<a name="requirements"></a>
+## Requirements
+
+This is a <span class="badge badge-info">PRO</span> operation. It requires that you have [purchased access to `backpack/pro`](https://backpackforlaravel.com/products/pro-for-unlimited-projects).
+
+In addition, it needs:
+- Your entity must use Laravel's `SoftDeletes` trait, for this Operation to work
+
+<a name="trash-a-single-item"></a>
+## Trash a Single Item <span class="badge badge-info">PRO</span>
+
+<a name="how-it-works"></a>
+### How it Works
+- **Trash**:
+Using AJAX, a DELETE request is performed towards ```/entity-name/{id}/trash```, which points to the ```trash()``` method in your EntityCrudController.
+
+- **Restore**:
+Using AJAX, a PUT request is performed towards ```/entity-name/{id}/restore```, which points to the ```restore()``` method in your EntityCrudController.
+
+- **Delete**:
+Using AJAX, a DELETE request is performed towards ```/entity-name/{id}/delete-permanently```, which points to the ```deletePermanently()``` method in your EntityCrudController.
+
+<a name="enabling"></a>
+### How to Use
+
+To enable it, you need to ```use \Backpack\PRO\Http\Controllers\Operations\TrashOperation;``` on your EntityCrudController. For example:
+
+```php
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+
+class ProductCrudController extends CrudController
+{
+    use \Backpack\PRO\Http\Controllers\Operations\TrashOperation;
+}
+```
+
```

Comment:
This operation like other operations expose some access permissions that you can configure. 

When used, `TrashOperation` exposes `trash`, `restore` and `delete_permanently` access for each corresponding operation. 
Likewise, `BulkTrashOperation` has `bulkTrash`, `bulkRestore` and `bulkDeletePermanently`. 

You can for example, revoke access to some operations depending on your user roles:
```php
// if user is not superadmin, don't allow permanently delete
public function setupTrashOperation() 
{
    if(! backpack_user()->hasRole('superadmin') {
         $this->crud->denyAccess('delete_permanently');
    }
}
```

You can also disable the default trash filter:
```php
public function setupTrashOperation() 
{
    $this->crud->setOperationSetting('withTrashFilter', false);
}
```
Disabling the filter also show the trashed items in your List view and configure the buttons according to the entry delete state. 
By default the `Delete Permanently` button is only shown in trashed items, if you want to allow your admins to permanently delete without sending first to trash you can achieve that by defining in your operation setup:

```php
// in the setupTrashOperation method
CRUD::setOperationSetting('allowDeleteWithoutTrash', true);
```